### ISSUE TITLE
DateTime::Format::Strptime 1.76 throws exception on invalid input when on_error => 'undef' was requested

### ISSUE THREAD

**Issue by x-qq** [houseabsolute/DateTime-Format-Strptime#25]
```
#!/usr/bin/env perl

use 5.026;
use strict;
use warnings;

use Try::Tiny;
use DateTime::Format::Strptime; # 1.76

my $bad_date_str = "2019-12-30 16:57:15+9999";

my $dt_parser = DateTime::Format::Strptime->new(
    time_zone => "UTC",
    pattern   => "%Y-%m-%d %H:%M:%S%z",
    strict    => 1,
    on_error  => "undef",
);

try {
    my $dt = $dt_parser->parse_datetime($bad_date_str);
    say "successfully alive here";
} catch {
    chomp;
    warn "EXCEPTION: " . $_;
}
```

```
% ./dtbr.pl  
EXCEPTION: Invalid offset: +9999 at ./dtbr.pl line 24.
%
```

Debian GNU/Linux Unstable amd64
perl 5.26.2

**Comment by autarch** [comment]
Thanks for the report. This is definitely a bug. Fortunately it's an easy fix. I'll try to get to this this weekend, but a PR would be welcome. The fix is to wrap all calls to `DateTime::TimeZone->new` in a `try` block. If it fails we should call `$self->_our_croak` with an appropriate error message.

**Comment by autarch** [comment]
Fixed in 1.77
